src/main/plugins/websearch-plugin/websearch-plugin.ts

```diff
@@ -7,6 +7,7 @@ import { WebSearchEngine } from "./web-search-engine";
 import { TranslationSet } from "../../../common/translation/translation-set";
 import { defaultWebSearchIcon } from "../../../common/icon/default-icons";
 import { isValidIcon } from "../../../common/icon/icon-helpers";
+import axios from "axios";
```

Comment:
I don't want the websearch plugin to depend on axios. Can you somehow abstract this so it can be unit tested in the future?

Reply:
I can use https that comes with node.

Reply:
I can use the https module that comes with nodejs but I will have to write a function to promisify its GET method.

Reply:
That's not what I mean. When doing actual HTTP requests axios is fine. When this plugin depends on axios or any other http client we can't unit test it's functionality without making actual HTTP requests. This is something I want to avoid. I would rather abstract the component that fetches suggestions, so we're able to mock it when unit testing.

Reply:
For example, take a look at the `urlExecutor` within this class. The plugin itself does not know how the implementation of the execution of URLs is implemented (because it differs depending on which operating system you are). We inject the actual url executor from outside into this class.

See here: https://github.com/oliverschwendener/ueli/blob/5377020897af4992cbe0fe4f1546f9ba5122319c/src/main/plugins/websearch-plugin/websearch-plugin.ts#L11-L21

Reply:
I see what you mean. But I don't know how can I resolve searchResults outside of axios call.
If ueli has some sorta of an updateResults method that I trigger when axios call finishes and updates the results shown to user.

Reply:
How about something like this?

``` typescript
public getSearchResults(userInput: string, fallback?: boolean): Promise<SearchResultItem[]> {
    return new Promise((resolve, reject) => {
        const suggestionPromises = this.config.webSearchEngines.map((webSearchEngine) => {
            return this.suggestionResolver(webSearchEngine.suggestionUrl);
        });

        Promise.all(suggestionPromises)
            .then((results) => resolve(results));
            .catch((error) => reject(error))
    });
}
```

Reply:
I already made this in my last commit except wrapping it in a suggestionResolver function but I resolve the results whether whether `Promise.all` is successful or not, because if one of the suggestionPromises fails the whole plugin would fail. 
So I abstracted the default results outside of Promise.all that way if anything goes wrong I can always show the user the default results, what do you think ? should I keep it this way or should I reject the whole plugin's results if the GET call fails ?

I will make another commit to clean up the code into a suggestionResolver function.

Reply:
Check my last commit where I cleaned up the code:
suggestionResolver function that returns a promise with suggestionsResults and if a webSearchEngine has no suggestionUrl or the GET request fails, the function returns empty promise. That way the